src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
Good points - I'm a bit torn on this myself.

At the point we might want to use other metrics that the jruby-metrics service is already tracking to decide how to enforce request limits, I'd agree with you that it would seem better to avoid having to duplicate any of the metrics that the jruby-metrics service tracks.

As a minor point, the jruby-metrics service currently does not have a tidy counter for tracking the number of "active requests".  You could probably capture that by summing a count of `requested-instances` and `borrowed-instances` from the metrics registry together.  It might be a bit less expensive, though, to just have a running active request counter local to the `wrap-with-concurrent-request-limit` which is checked.  It seems like the logic wouldn't have to be too involved to do that.  I'm thinking of something like this:

```clojure
(defn wrap-with-concurrent-request-limit
  [handler metrics-service request-limit splay-limit]
  (let [request-count (atom 0)]
    (fn [request]
      (swap! request-count inc)
      (try
        ...
        (finally
          (swap! request-count dec))))))
```

For now, I'd probably still lean toward doing something like the above for now to avoid having to tie this functionality to the presence / enablement of the metrics service - and the associated error handling that we'd have to write around that as a result.

Would be interested to get another opinion on this, though.  Maybe @jpinsonault or @rlinehan could weigh in here.